dbmongo/lib/engine/typescript.go

```diff
@@ -4,15 +4,52 @@ import (
 	"log"
 	"os"
 	"os/exec"
+	"path"
 	"path/filepath"
+	"strings"
 )
 
-func TranspileTsFunctions(jsRootDir string) {
-	// TODO: also transpile any other TS files
-	tsFiles := []string{
-		filepath.Join(jsRootDir, "common", "raison_sociale.ts"),
-		filepath.Join(jsRootDir, "reduce.algo2", "fraisFinancier.ts"),
+func shouldTranspile(filePath string) bool {
+	return !strings.Contains(filePath, "node_modules") &&
+		!strings.Contains(filePath, "_tests.ts") &&
+		path.Ext(filePath) == ".ts"
+}
+
+// ListTsFiles retourne la liste des fichiers TypeScript transpilable en JavaScript
+// en cherchant récursivement depuis le répertoire jsRootDir.
+func ListTsFiles(jsRootDir string) []string {
+	var files []string
+	err := filepath.Walk(jsRootDir, func(filePath string, info os.FileInfo, err error) error {
+		if err == nil && shouldTranspile(filePath) {
+			files = append(files, filePath)
+		}
+		return err
+	})
+	if err != nil {
+		log.Fatal(err)
 	}
+	return files
+}
+
+// DeleteTranspiledFiles supprime les fichiers JavaScript résultant de la
+// transpilation des fichiers TypeScript listés dans tsFiles.
+func DeleteTranspiledFiles(tsFiles []string) {
+	for _, tsFile := range tsFiles {
+		ext := path.Ext(tsFile)
+		if ext != ".ts" {
+			panic("expected a .ts file, found: " + tsFile)
+		}
+		transpiledFile := tsFile[0:len(tsFile)-len(ext)] + ".js"
```

Comment:
Peut-être un peu plus lisible avec:
```suggestion
		transpiledFile := strings.TrimSuffix(tsFile, ext) + ".js"
```

Reply:
stylé ! merci Pierre !